### PR TITLE
Components Sprint 8 - Part 4

### PR DESCRIPTION
**Pull request by nelson-glauber**
This PR includes the following changes:
- Added `DismissibleNavigationDrawer` and `PermanentNavigationDrawer`
- Added `DismissibleDrawerSheet` and `PermanentDrawerSheet `
```
<DismissibleNavigationDrawer is-open={"#{@drawerIsOpen}"} on-close="closeDrawer" on-open="openDrawer">
  <DismissibleDrawerSheet template="drawerContent">
    <NavigationDrawerItem selected="true" phx-click="navigateToGrid">
      <Text template="label">Grid</Text>
    </NavigationDrawerItem>
    ...
  </DismissibleDrawerSheet>
  <Scaffold> 
  ...
  </Scaffold>
</DismissibleNavigationDrawer>
```
- Added `DropdownMenu` composable.
```
<IconButton template="action" phx-click="showPopup">
  <Icon image-vector="filled:MoreVert" />
  <DropdownMenu phx-click="hidePopup" expanded={"#{ @showPopup }"}>
    <DropdownMenuItem phx-click="onMenuOptionClick" phx-value="A">
      <Text>Option A</Text>
    </DropdownMenuItem>
    <DropdownMenuItem phx-click="onMenuOptionClick" phx-value="B">
      <Text>Option B</Text>
    </DropdownMenuItem>
  </DropdownMenu>
</IconButton>
```
- Added `Surface` composable.
```
<Surface shape="12" color="system-blue" content-color="system-white"
  border="{'width': '2', 'color': 'system-green'}" phx-click="onClick">
  <Text padding="32">Surface</Text>
</Surface>
```

### CODE REVIEW

**Comment by AZholtkevych** [comment]
@carson-katri could you please help here with the review?

File: liveview-android/src/main/java/org/phoenixframework/liveview/data/dto/SurfaceDTO.kt [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+ *   <Text padding="16">Simple Surface</Text>
+ * </Surface>
+ *
+ * <Surface shape="12" color="system-blue" content-color="system-white"
+ *   phx-click="onClick">
+ *   <Text padding="32">Clickable Surface</Text>
+ * </Surface>
+ *
+ * <Surface shape="12" color="system-blue" content-color="system-white"
+ *   selected={"#{@isSelected}"} phx-click="selectItem"} >
```
Comment (carson-katri) [inline liveview-android/src/main/java/org/phoenixframework/liveview/data/dto/SurfaceDTO.kt:64]:
```suggestion
 *   selected={"#{@isSelected}"} phx-click="selectItem">
```

File: liveview-android/src/main/java/org/phoenixframework/liveview/data/dto/DropdownMenuDTO.kt [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+ * def handle_event("showPopup", _params, socket) do
+ *   {:noreply, assign(socket, :showPopup, true)}
+ * end
+ *
+ * def handle_event("hidePopup", _params, socket) do
+ *   {:noreply, assign(socket, :showPopup, false)}
+ * end
+ *
+ * def handle_event("onMenuOptionClick", _params, socket) do
+ *   {:noreply, socket |> assign(:menuOption, _params) |> assign(:showPopup, false) }
```
Comment (carson-katri) [inline liveview-android/src/main/java/org/phoenixframework/liveview/data/dto/DropdownMenuDTO.kt:44]: nit: I think an underscore typically indicates the value will not be used in the function body.

File: liveview-android/src/main/java/org/phoenixframework/liveview/data/dto/SurfaceDTO.kt (hunk 1 above)
Comment (carson-katri) [reply liveview-android/src/main/java/org/phoenixframework/liveview/data/dto/SurfaceDTO.kt]: This line still has an extra `}`

File: liveview-android/src/main/java/org/phoenixframework/liveview/data/dto/SurfaceDTO.kt [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+ *   <Text padding="32">Selectable Surface</Text>
+ * </Surface>
+ *
+ * <Surface shape="12" color="system-blue" content-color="system-white"
+ *   checked={"#{@isChecked}"} phx-change="toggleCheck"} >
```
[the remark this answers is not in the document]
Comment (carson-katri) [reply liveview-android/src/main/java/org/phoenixframework/liveview/data/dto/SurfaceDTO.kt]: This line still has an extra `}`

**Review by carson-katri — APPROVED** [review]

**Pull request merged**